### PR TITLE
[Need review] Re-factor store commands

### LINKED ISSUE #310: Refactoring: store commands should be in their own .go files
## Expected Behaviour

store commands should be in separate files like we have for `template pull`

## Current Behaviour

They're intermingled in the same file.

## Possible Solution

Please split out so we retain the coding style.

I'd suggest that @itscaro may want to work on this?

screen shot 2018-01-31 at 21 19 54

### PR DESCRIPTION
**Pull request by itscaro**
## Description
This PR corrects also: `store deploy` must not consider yaml file

## Motivation and Context

- [x] I have raised an issue to propose this change (Closes #310)

## How Has This Been Tested?
I tested these commands:

* faas-cli store list
* faas-cli store list -u "https://cdn.rawgit.com/openfaas/store/master/store-arm64.json"
* faas-cli store inspect "Tesseract OCR"
* faas-cli store inspect -u "https://cdn.rawgit.com/openfaas/store/master/store-arm64.json" sha512sum
* faas-cli store deploy Figlet -g http://localhost:8081
* faas-cli.exe store deploy "Tesseract OCR" --replace=true --update=false --env=openfaas.label=itscaro
* faas-cli.exe store deploy "Tesseract OCR" --replace=true --update=false

## Types of changes

- [x] Re-factor
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: commands/store.go [hunk 1]
```diff
@@ -247,14 +106,14 @@ func storeList(store string) ([]schema.StoreItem, error) {
...
 		bytesOut, err := ioutil.ReadAll(res.Body)
 		if err != nil {
-			return nil, fmt.Errorf("cannot read result from OpenFaaS store on URL: %s", store)
+			return nil, fmt.Errorf("cannot read result from OpenFaaS store at URL: %s", store)
 		}
+
 		jsonErr := json.Unmarshal(bytesOut, &results)
 		if jsonErr != nil {
-			return nil, fmt.Errorf("cannot parse result from OpenFaaS store on URL: %s\n%s", store, jsonErr.Error())
+			return nil, fmt.Errorf("cannot parse result from OpenFaaS store at URL: %s\n%s", store, jsonErr.Error())
```
Comment (alexellis) [inline commands/store.go]: 👍

File: commands/store_deploy.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
```
Comment (alexellis) [inline commands/store_deploy.go]: Move it to 2018?

**Comment by alexellis** [comment]
LGTM, see comments: 

1.

> How Has This Been Tested?

This field cannot not be left empty. We should be setting an example as maintainers on the CLI.

2. do we need a follow-up PR to validate the environment is passed correctly with a `=` symbol?

File: commands/store_deploy.go (hunk 2 above)
Comment (itscaro) [reply commands/store_deploy.go]: ok

**Comment by itscaro** [comment]
> How Has This Been Tested?

I will update with capture of command testing 

> do we need a follow-up PR to validate the environment is passed correctly with a = symbol?

No I don't think so, since this seems correctly handled in `parseMap()` in `deploy.go`, maybe a PR for unit-test on that util func

File: commands/store_deploy.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			storeDeployFlags.labelOpts = append(storeDeployFlags.labelOpts, label)
+		}
+	}
+
+	// Use the network from manifest if not changed by user
+	if !cmd.Flag("network").Changed {
+		network = item.Network
+	}
+
+	return deployImage(
```
Comment (itscaro) [inline commands/store_deploy.go]: it was RunDeploy()
I discovered a bug when in the working directory we have stack.yml, the `store deploy` use the same function in `deploy` which gives priority to yaml file. In the context of store, yaml file should not be used, that's why I re-factored this part

File: commands/deploy.go [hunk 4]
```diff
@@ -197,19 +197,34 @@ func RunDeploy(
 			return fmt.Errorf("please provide a --name for your function as it will be deployed on FaaS")
 		}
 
-		envvars, err := parseMap(deployFlags.envvarOpts, "env")
-		if err != nil {
-			return fmt.Errorf("error parsing envvars: %v", err)
+		if err := deployImage(image, fprocess, functionName, deployFlags); err != nil {
```
Comment (itscaro) [inline commands/deploy.go]: re-factoring this part so that `store deploy` can call directly this code instead of passing by `RunDeploy()`

**Comment by alexellis** [comment]
I'll be looking at this when I'm back next week @itscaro - in the meantime maybe @ericstoekl could take a look?

**Comment by alexellis** [comment]
Hey @itscaro I haven't seen any reviews on this yet, but how is the testing going?

> itscaro: I will update with capture of command testing

**Comment by itscaro** [comment]
@alexellis hey Alex, I updated the PR description to add use-case that I tested locally

**Comment by itscaro** [comment]
@alexellis hey Alex, I updated the PR description to add use-case that I tested locally, just updated to add deploy with some flags

File: commands/store_deploy.go (hunk 3 above)
Comment (alexellis) [reply commands/store_deploy.go]: We have the same issue with deploying via the CLI only when a stack.yml file is there. https://github.com/openfaas/faas-cli/pull/334#issuecomment-369023174

**Comment by alexellis** [comment]
Change is quite large but I see it's been open for at least a couple of weeks and the tests pass.. so let's get it in.

**Comment by alexellis** [comment]
Thanks for working on this @itscaro

**Pull request merged**